### PR TITLE
feat: add repr to lr schedulers

### LINKED ISSUE #168967: implement __str__ for Schedular classes
### 🚀 The feature, motivation and pitch

I'm logging my setup to a txt file, including model/optimizer/scheduler, and would like the string representation of schedulers to be meaningful, similar to `nn.Module` and Pytorch's Optimizers.

### Alternatives

_No response_

### Additional context

logging a scheduler: 
```
<torch.optim.lr_scheduler.OneCycleLR object at 0x14b688123e30>
```
, which is not meaningful, while logging the optimizers
```
AdamW (
Parameter Group 0
    amsgrad: False
    base_momentum: 0.85
    betas: (0.95, 0.999)
    capturable: False
    decoupled_weight_decay: True
    differentiable: False
    eps: 1e-08
    foreach: None
    fused: None
    initial_lr: 4e-05
    lr: 3.9999999999999996e-05
    max_lr: 0.001
    max_momentum: 0.95
    maximize: False
    min_lr: 4e-09
    weight_decay: 0.01
)
```

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

### PR DESCRIPTION
**Pull request by tshauck**
Fixes #168967

```
In [1]: import torch

In [2]:     class SchedulerTestNet(torch.nn.Module):
   ...:         def __init__(self) -> None:
   ...:             super().__init__()
   ...:             self.conv1 = torch.nn.Conv2d(1, 1, 1)
   ...:             self.conv2 = torch.nn.Conv2d(1, 1, 1)
   ...: 
   ...:         def forward(self, x):
   ...:             return self.conv2(F.relu(self.conv1(x)))
   ...: 

In [3]: net = SchedulerTestNet()

In [4]: import torch.optim as optim

In [5]: opt = optim.SGD(
   ...:     [
   ...:         {"params": net.conv1.parameters()},
   ...:         {"params": net.conv2.parameters(), "lr": torch.tensor(0.5)},
   ...:     ],
   ...:     lr=0.05
   ...:     )

In [6]: scheduler = optim.lr_scheduler.StepLR(opt, gamma=0.1, step_size=3)

In [7]: print(scheduler)
StepLR (
    step_size: 3
    gamma: 0.1
    last_epoch: 0
    base_lrs: [0.05, 0.5]
)
```

### CODE REVIEW

**Review by janeyx99 — APPROVED** [review]
Thanks!

**Comment by janeyx99** [comment]
@claude can you also make suggested edits to change all the extra_repr defs into private defs `_extra_repr`
